src/Helpers/RemoteFile.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Spatie\MediaLibrary\Helpers;
+
+class RemoteFile
+{
+    /**
+     * The relative path to the file.
+     *
+     * @var string
+     */
+    protected $key;
+
+    /**
+     * The disk the file exists on.
+     *
+     * @var string
+     */
+    protected $disk;
+
+    /**
+     * Constructor method.
+     *
+     * @return void
+     */
+    public function __construct($key, $disk)
+    {
+        $this->key = $key;
+        $this->disk = $disk;
+    }
+
+    /**
+     * Get the key.
+     *
+     * @return string
+     */
+    public function getKey(): string
+    {
+        return $this->key;
+    }
+
+    /**
+     * Get the disk.
+     *
+     * @return string
+     */
+    public function getDisk(): string
+    {
+        return $this->disk;
+    }
+
+    /**
+     * Get the filename (including extension).
+     *
+     * @return string
+     */
+    public function getFilename()
+    {
+        return basename($this->key);
+    }
+
+    /**
+     * Get the name (excluding extension).
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return pathinfo($this->getFilename())['filename'];
```

Comment:
Use the second parameter of pathinfo to get the filename instead of retrieving a key from the result array.